### ISSUE TITLE
Extend STS Session Time

### ISSUE THREAD

**Issue by garyrobert** [oktadeveloper/okta-aws-cli-assume-role#108]
AWS recently updated their policy to allow for STS tokens up to 12 hours.

https://aws.amazon.com/about-aws/whats-new/2018/03/longer-role-sessions/

There are two changes that are required to use extended session tokens. 

1) An IAM admin in the AWS console will need to extend the Maximum CLI/API session duration for the IAM role created for Okta.

![image](https://user-images.githubusercontent.com/5948803/38144133-6bed0414-33f8-11e8-9230-d157875ae399.png)

2) When generating the STS via CLI, the user will need to supply a duration-seconds parameter to increase the 1 hour STS token.

Since the okta-aws-cli-assume-role tool creates STS tokens with a 1 hour session, I'm requesting that the tool be updated to allow a user to supply a flag to extend the session up to 12 hours which in turn will get passed to the `aws sts` command. Similar to the `awscli logout` command.

For example: `awscli duration-seconds 43200`

**Comment by jonjohnston** [comment]
Yes this is a very important option for our company. We have been waiting for Amazon to release this. Hoping this can be implemented into the okta-aws-cli-assume-role ASAP

**Comment by AlainODea** [comment]
@jonjohnston the way the tool is currently written it keeps refreshing the Okta session and updating the derived AWS profiles. As long as you periodically call `withokta $profilename true` you'll keep that session usable by other code indefinitely.

My experience is that the timeouts are set to the timeout configured on the Amazon Web Services app in my Okta Org.

@garyrobert given that the tool automatically refreshes sessions and profiles, do you still want longer sessions?

**Comment by slancio** [comment]
Yes, this really needs to be at least configurable.  We have long-running jobs we want to be able to run where periodically running withokta is kind of undesirable.  We can make do with that but making the AWS features is better.

**Comment by AlainODea** [comment]
Do any of you write Java? This is a pretty straight-forward change and someone motivated by necessity would deliver it more effectively.

I can provide some guidance on how to implement this if someone is willing to do the coding and the testing.

**Comment by AlainODea** [comment]
Full disclosure, I don’t work for Okta. Most of the recent contributions to this project (including mine) are community contributions (from outside Okta) and that makes for a healthy project.

**Comment by slancio** [comment]
I'm a developer, though not a Java one.  It's fairly easy to change the number for the timeout, but I'm happy to hear what a preferred solution might look like and take a try at it.

**Comment by AlainODea** [comment]
@slancio thank you!

I recommend the following approach (the names are up to you):
1. Add support for a new **config.properties** setting. Add something like **OKTA_STS_DURATION** to **OktaAwsConfig** similar to how **OKTA_PROFILE** is done here: 
https://github.com/oktadeveloper/okta-aws-cli-assume-role/blob/master/src/main/java/com/okta/tools/OktaAwsConfig.java#L32
2. Add a field to the common environment **OktaAwsCliEnvironment**. You could add something like **stsDuration** as a field
3. Use **OktaAwsCliEnvironment** to hand that value to the **AssumeRoleWithSAMLRequest**:
https://github.com/oktadeveloper/okta-aws-cli-assume-role/blob/b2c6ba8927ffc0991a76e2b8c31db9001570ef01/src/main/java/com/okta/tools/helpers/RoleHelper.java#L96-L100

**Comment by slancio** [comment]
https://github.com/oktadeveloper/okta-aws-cli-assume-role/pull/111  This is slightly more involved than suggested, but let me know if it needs more work.

**Comment by AlainODea** [comment]
@andreihava-okta this can be closed. It is fixed.
